### ISSUE TITLE
Check for unsupported default-overwrites from Astropy Tables

### ISSUE THREAD

**Issue by tbowers7** [pypeit/PypeIt#1330]
In Astropy 5.0, support was removed for the deprecated `overwrite=None` option for `astropy.io.ascii.ui.write()`.  Overwriting existing files now only happens if `overwrite=True` is specified in the call to `.write()`.

One such instance in the PypeIt code base was corrected in #1329 within the `pypeit_identify` script.  As users upgrade to Astropy 5.0+, they may encounter other places in the PypeIt code base where the default-overwrite behavior is still expected.

A quick grep of the `pypeit` directory shows a number of places where `.write()` is called without specifying the `overwrite` argument, but it is unclear which of these situations _expect_ to overwrite an existing file and which not.

---
From the Astropy Changelog:

**_5.0 (2021-11-15)_**
**astropy.io.ascii**
Removed deprecated `overwrite=None` option for `astropy.io.ascii.ui.write()`. Overwriting existing files now only happens if `overwrite=True`. [#12171]

**_1.3 (2016-12-22)_**
**astropy.io.ascii**
ASCII writers now accept an `overwrite` argument. The default behavior is changed so that a warning will be issued when overwriting an existing file unless `overwrite=True`. In a future version this will be changed from a warning to an exception to prevent accidentally overwriting a file. [#5007]

**Comment by tbowers7** [comment]
Working purely from a grep of "`.write(`" in the `pypeit` directory (and excluding the `deprecated` tree), I find three groups of instances related to this issue:

1. `ascii.write()` instances that already have an `overwrite` keyword call.
2. `ascii.write()` instances that might need an `overwrite` call added, depending on context.
3. Possible FITS table writers (uncertain whether we need `overwrite` here or not).

Filename/line number, and line of code for each instance is listed below, separated by case.

---
1. `ascii.write()` instances that already have an `overwrite` keyword call.  (Note: `pypeit/core/gui/identify.py` is included in #1329.)
```
pypeit/core/gui/identify.py:1249:        ascii_io.write(data, fname, format='fixed_width', overwrite=True)
pypeit/core/wavecal/templates.py:608:        vac_list.write(new_file, format='ascii.fixed_width', overwrite=True)
pypeit/specobjs.py:873:            obj_tbl.write(outfile,format='ascii.fixed_width', overwrite=True)
pypeit/data/filters/mk_filtercurve.py:290:    tbl.write('filter_list.ascii', format='ascii', overwrite=True)
```

2. `ascii.write()` instances that might need an `overwrite` call added, depending on context.
```
pypeit/wavemodel.py:715:        dat.write(file_root_name+'_lines.dat',format='ascii.fixed_width')
pypeit/metadata.py:1446:            subtbl.write(ff, format='ascii.fixed_width')
pypeit/metadata.py:1626:                subtbl.write(ff, format='ascii.fixed_width')
pypeit/metadata.py:1756:            output_tbl.write(ff, format='ascii.fixed_width')
pypeit/core/wavecal/waveio.py:531:        tbl.write(f, format='ascii.fixed_width')
pypeit/scripts/ql_keck_deimos.py:216:        ps.usrdata.write(ff, format='ascii.fixed_width')
pypeit/data/arc_lines/generate.py:140:    subt.write(outfil, delimiter='|',format='ascii.fixed_width')
pypeit/data/arc_lines/convert_NIST_to_lists.py:195:        tbl.write(f, format='ascii.fixed_width')
pypeit/archive.py:140:                ascii.write(t, f, format='ipac')
```

3. Possible FITS table writers (uncertain whether we need `overwrite` here or not).
```
pypeit/core/wavecal/wvutils.py:714:    tbl.write(outfile, overwrite=overwrite)
pypeit/core/wavecal/templates.py:542:        tbl.write(outfile, overwrite=True)
pypeit/core/wavecal/templates.py:594:            tbl.write(outfile, overwrite=True)
pypeit/core/wavecal/templates.py:637:        tbl.write(outfile, overwrite=True)
pypeit/core/wavecal/templates.py:667:        tbl.write(outfile, overwrite=True)
pypeit/core/wavecal/templates.py:707:        tbl.write(outfile, overwrite=True)
pypeit/core/wavecal/templates.py:795:        tbl.write(outfile, overwrite=True)
pypeit/scripts/obslog.py:145:        fitstbl = ps.fitstbl.write(output='table' if args.interact else _file, rows=indx,
pypeit/scripts/flux_setup.py:90:            fin.write(data)
pypeit/scripts/flux_setup.py:120:            fin.write(data)
pypeit/pypeitsetup.py:496:            self.fitstbl.write(output=log_file, columns='pypeit', sort_col='mjd',
```

**Comment by profxj** [comment]
Gack.  Good catch.  Myself, I wish we would
just use `pandas`, but not going to happen given
that we are wedded to FITS.

We'll discuss this Issue at the next Team meeting.

**Comment by tbowers7** [comment]
Fixed in #1360 :
```
pypeit/core/wavecal/waveio.py:531:        tbl.write(f, format='ascii.fixed_width')
```

**Comment by kbwestfall** [comment]
The last comment indicates this is fixed.  Closing.
